x-pack/legacy/plugins/code/server/indexer/lsp_indexer.ts

```diff
@@ -105,6 +108,8 @@ export class LspIndexer extends AbstractIndexer {
 
   protected async *getIndexRequestIterator(): AsyncIterableIterator<LspIndexRequest> {
     try {
+      await this.prepareWorkspace();
+
       const fileIterator = await this.gitOps.iterateRepo(this.repoUri, HEAD);
```

Comment:
in this PR, we're iterating files using local file system, there are potential problems we need to consider later
1. if workspace couldn't be created, then index will fail (less of problem because it's hard for native git to fail on create workspace)
2. during the index, it require language server not modify files in the workspace, while it's a reasonable request, it may not be very easy to gurantee this
3. for git file that is a link, we need to figure out the behavior in different platform, and make sure we don't index file content of the linked file

Reply:
+1 on this. We might want to add these as follow up discussion items after this PR has been merged.